src/google-classroom-extractor/google_classroom_extractor/main.py

```diff
@@ -0,0 +1,222 @@
+        [
+            "courseWorkId",
+            "courseWorkState",
+            "courseWorkTitle",
+            "courseWorkDescription",
+            "courseWorkCreationTime",
+            "courseWorkUpdateTime",
+            "courseWorkDueDate",
+        ]
+    ]
+
+    course_coursework_submissions_df: pd.DataFrame = pd.merge(
+        course_submissions_df, coursework_df, on="courseWorkId", how="left"
+    )
+
+    students_df.rename(
+        columns={
+            "profile.name.fullName": "studentName",
+            "profile.emailAddress": "studentEmail",
+            "userId": "studentUserId",
+        },
+        inplace=True,
+    )
+    students_df = students_df[
+        ["studentUserId", "courseId", "studentName", "studentEmail"]
+    ]
+
+    full_df: pd.DataFrame = pd.merge(
+        course_coursework_submissions_df,
+        students_df,
+        on=["studentUserId", "courseId"],
+        how="left",
+    )
+
+    full_df["importDate"] = datetime.today().strftime("%Y-%m-%d")
+
+    return full_df
+
+
+def request():
+    load_dotenv()
+    logging.basicConfig(
+        handlers=[
+            logging.StreamHandler(sys.stdout),
+        ],
+        format="%(asctime)s - %(levelname)s - %(message)s",
+        level=os.environ.get("LOGLEVEL", "INFO"),
+    )
+    logging.getLogger("matplotlib").setLevel(logging.ERROR)
+    logging.getLogger("opnieuw").setLevel(logging.DEBUG)
+
+    credentials: service_account.Credentials = get_credentials()
+    reports_resource: Resource = build(
+        "admin", "reports_v1", credentials=credentials, cache_discovery=False
+    )
+    classroom_resource: Resource = build(
+        "classroom", "v1", credentials=credentials, cache_discovery=False
+    )
+
+    sync_db = get_sync_db_engine()
+
+    courses_df: pd.DataFrame = request_all_courses_as_df(classroom_resource, sync_db)
+    course_ids: List[str] = courses_df["id"].tolist()
+
+    coursework_df: pd.DataFrame = request_all_coursework_as_df(
+        classroom_resource, course_ids, sync_db
+    )
+    submissions_df: pd.DataFrame = request_all_submissions_as_df(
+        classroom_resource, course_ids, sync_db
+    )
+    students_df: pd.DataFrame = request_all_students_as_df(
+        classroom_resource, course_ids, sync_db
+    )
+
+    Result = collections.namedtuple(
+        "Result",
+        [
+            "usage",
+            "normalized_submissions",
+            "courses",
+            "coursework",
+            "submissions",
+            "students",
+        ],
+    )
+    return Result(
+        lambda: request_all_usage_as_df(reports_resource, sync_db),
+        lambda: normalized_submissions(
+            courses_df=courses_df,
+            coursework_df=coursework_df,
+            submissions_df=submissions_df,
+            students_df=students_df,
+        ),
+        lambda: courses_df,
+        lambda: coursework_df,
+        lambda: submissions_df,
+        lambda: students_df,
+    )
+
+
+if __name__ == "__main__":
```

Comment:
Instead of executing the same script from both places, I would just use the main for generating the csv and use each notebook to consume the internal packages and include some logic in them, so the notebooks and the main could be used as examples of how to consume the packages.

Reply:
Sounds good.  Looking at this now, I should also split pulling data from getting data out of sync db.